src/systemctl/systemctl.c

```diff
@@ -5959,6 +6004,18 @@ static int cat(int argc, char *argv[], void *userdata) {
         if (r < 0)
                 return log_error_errno(r, "Failed to expand names: %m");
 
+        if (arg_with_dependencies) {
+                _cleanup_strv_free_ char **names_with_deps = NULL;
+
+                r = append_unit_dependencies(bus, names, &names_with_deps);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to append unit dependencies: %m");
+
+                strv_free(names);
+                names = TAKE_PTR(names_with_deps);
+        }
```

Comment:
this code you add in a pretty similar way a bit earlier. Maybe convert this into a function of its own:

```c
static int maybe_extend_with_unit_dependencies(sd_bus *bus, char ***list) { 
…
```

and that function would leave the specified list untouched if arg_with_dependencies is false, and otherwise augment it with the dependencies, updating the specified argument in place.